src/comments.js

```diff
@@ -412,19 +416,27 @@ function printComments(path, print, options) {
             parent
           )
         );
+      } else {
+        // Those two node types print their dangling comments themselves
+        if (value.type !== "Program" && value.type !== "BlockStatement") {
+          trailingParts.push(
+            printDanglingComment(
+              commentPath,
+              print,
+              options
+            )
+          );
+        }
```

Comment:
The problem is that you have to special-case all the places that have *already* emitted comments, which just doesn't seem worth it to me for all that complexity. Also, we could potentially emit a line comment in the wrong place and produce invalid JS code.

I get the idea of wanting to at least keep comments around so they aren't lost, but dangling comments are pretty rare in the real-world anyway. There are several more places we should manually emit them and it feels wrong to embed that list here to avoid printing them automatically for those.